### PR TITLE
remove project-wide deployment settings

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
There were some project wide deployment settings that were causing more harm than good given how datatools is used these days! These settings are removed in this PR

### CODE REVIEW

**Review by br648 — APPROVED** [review]
All good.

File: src/main/java/com/conveyal/datatools/manager/models/Deployment.java [hunk 1]
```diff
@@ -442,22 +442,12 @@ public byte[] generateBuildConfig() throws IOException {
...
-        Project project = parentProject();
-        return project.buildConfig != null
-            ? writeToBytes(project.buildConfig)
             : null;
     }
 
     public String generateBuildConfigAsString() {
         if (customBuildConfig != null) return customBuildConfig;
-        return writeToString(this.parentProject().buildConfig);
+        return null;
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/Deployment.java]: Simplify to `return customBuildConfig;`. You might even be able to inline this method.

File: src/main/java/com/conveyal/datatools/manager/models/OtpBuildConfig.java [hunk 2]
```diff
@@ -11,23 +11,4 @@
 public class OtpBuildConfig implements Serializable {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/OtpBuildConfig.java:13]: This is now an empty class, can we remove it without messing up the loading of existing datatools projects?

File: src/main/java/com/conveyal/datatools/manager/models/OtpRouterConfig.java [hunk 3]
```diff
@@ -11,31 +11,5 @@
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class OtpRouterConfig implements Serializable {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/OtpRouterConfig.java:14]: This is now an empty class, can we remove it without messing up the loading of existing datatools projects?

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks good with some further easy cleanup as indicated.

File: src/main/java/com/conveyal/datatools/manager/models/OtpRouterConfig.java (hunk 3 above)
Comment (miles-grant-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/models/OtpRouterConfig.java:14]: Removing it caused some problems with db migration, but I can add a TODO to remove these in the future

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**